Automattic/pocket-casts-ios
# What's new: Autoplay

This PR adds the "What's New" view for the Autoplay feature.

https://github.com/Automattic/pocket-casts-ios/assets/7040243/d194c679-2a2d-4764-bf7a-38837f1aae01

## To test

1. Run the app
2. Enable the `autoplay` feature flag (Settings > Beta Features)
3. Change `WhatsNew.viewControllerToShow()` to:

```swift
func viewControllerToShow() -> UIViewController? {
        let announcement = announcements.last!

        let whatsNewViewController = ThemedHostingController(rootView: WhatsNewView(announcement: announcement))
        whatsNewViewController.modalPresentationStyle = .overCurrentContext
        whatsNewViewController.modalTransitionStyle = .crossDissolve
        whatsNewViewController.view.backgroundColor = .init(red: 0, green: 0, blue: 0, alpha: 0.5)

        return whatsNewViewController
    }
```

4. Run the app again
5. You should see the popup
6. ✅ Tap "Maybe later"
7. The popup should dismiss
8. Re-run the app
9. Tap "Enable it"
10. ✅ Profile should show, then the general settings and then the screen should scroll to the Autoplay option

Test the "Enable it" flow from the popup appearing above Podcasts/Filters/Discover and above Profile, the flows are slightly different and they all should work.

## Checklist
